### PR TITLE
Submission Homework Week2

### PR DESCRIPTION
**Pull request by Mehmet-Ergorgec**
Part 3 is now in progress. I am going to do and commit it asap.

### CODE REVIEW

File: Week2/database.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+const getLanguagesInRegion = async () => {
+  try {
+    const response = await prompts({
+      type: 'text',
+      name: 'region',
+      message: `List all the languages spoken in the region ...?`,
+    });
+    const userInput = response.region.charAt(0).toUpperCase() + response.region.slice(1);
+    const regionQuery = await execQuery(question2, userInput);
+    const validateQuery = await execQuery(`select region from country group by region`);
```
Comment (mogaleaf) [inline Week2/database.js]: Why this query ? You could just check the length of regionQuery to see if you have language .

File: Week2/database.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+};
+
+const getLanguagesInRegion = async () => {
+  try {
+    const response = await prompts({
+      type: 'text',
+      name: 'region',
+      message: `List all the languages spoken in the region ...?`,
+    });
+    const userInput = response.region.charAt(0).toUpperCase() + response.region.slice(1);
```
Comment (mogaleaf) [inline Week2/database.js]: You don't need this, sql is not case sensitive.

File: Week2/database.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+const getTotalCityWithLanguage = async () => {
+  try {
+    const response = await prompts({
+      type: 'text',
+      name: 'language',
+      message: `Find out how many cities speak this language?`,
+    });
+    const userInput = response.language.charAt(0).toUpperCase() + response.language.slice(1);
+    const totalCitiesSpeakLanguage = await execQuery(question3, response.language);
+    const validateQuery = await execQuery(`select language from countrylanguage`);
```
Comment (mogaleaf) [inline Week2/database.js]: Same here no need, just check the length of the response.

File: Week2/database.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+const getOfficialOrNot = async () => {
+  const response = await prompts({
+    type: 'list',
+    name: 'value',
+    message: 'Are there any countries uses this language officially or not?',
+    initial: 'Ex:Middle East,Turkish',
+    separator: ',',
+  });
+  const isOfficial = await execQuery(question4, response.value);
+  const trueOfficial = isOfficial.filter(item => item.officialOrNot === 'T');
```
Comment (mogaleaf) [inline Week2/database.js]: this can be done inside the sql query to avoid work here.

File: Week2/queries.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
+const question1 = `
+SELECT
```
Comment (mogaleaf) [inline Week2/queries.js]: good

File: Week2/queries.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+const question2 = `
+   SELECT
+      countrylanguage.language AS language 
+   FROM
+      country,
+      countrylanguage 
+   WHERE
+      countrylanguage.countrycode = country.code 
+      AND country.region = ? 
+   GROUP BY
```
Comment (mogaleaf) [inline Week2/queries.js]: No need of group by here

File: Week2/queries.js [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+      language 
+   ORDER BY
+      language`;
+
+const question3 = `
+      SELECT
```
Comment (mogaleaf) [inline Week2/queries.js]: good

File: Week2/queries.js [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+        AND country.code = countrylanguage.countrycode 
+        AND countrylanguage.language = ? 
+      GROUP BY
+        language`;
+
+const question4 = `SELECT
+        region,
+        language,
+        country.name AS country,
+        isofficial as officialOrNot
```
Comment (mogaleaf) [inline Week2/queries.js]: Add the condition in the WHERE clause

File: Week2/queries.js [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+        countrylanguage
+     WHERE
+        countrylanguage.countrycode = country.code
+        AND region = ?
+        AND language = ? `;
+
+const question5 = `
+        SELECT
+           continent,
+           count(language) AS total_spoken_language
```
Comment (mogaleaf) [inline Week2/queries.js]: What if english is spoken in canada and USA and you choose America ? It will be counted twice

File: Week2/trigger.js [hunk 10]
```diff
@@ -0,0 +1,19 @@
+DELIMITER $$
+  CREATE TRIGGER max_language
```
Comment (mogaleaf) [inline Week2/trigger.js]: good

File: Week2/database.js (hunk 1 above)
Comment (Mehmet-Ergorgec) [reply Week2/database.js]: I thought something different. But you are right.Thank you

File: Week2/database.js (hunk 2 above)
Comment (Mehmet-Ergorgec) [reply Week2/database.js]: When I write this code which is Js , Mysql did not accept lower case. So I implemented this code. But I have changed it. Now no need. Thank you

File: Week2/queries.js (hunk 6 above)
Comment (Mehmet-Ergorgec) [reply Week2/queries.js]: There are some duplicated regions so I group them.

**Comment by Mehmet-Ergorgec** [comment]
Thank you for your  kind review. I implemented all your feedbacks. But in ERD Diagram each tag,reminder or mark should refer listItem instead of list. In homework I understand like that. I am looking forward to hear feedback about ERD because I will use it Homework week 3 after change some logic as well:)

File: Week2/queries.js (hunk 6 above)
Comment (mogaleaf) [reply Week2/queries.js]: Ok got it, you could have used a distinct also on countrylanguage .

**Comment by Mehmet-Ergorgec** [comment]
I have not inserted any listItem. I just created tables.because of that you did not see any listItem.  I understand that if I dont have any listItem in my list there is no need for reminder and tag. After I insert listItem in my list it is more meaningful to have reminder and tag.
If it is so, should I create my reminder,tag and mark tables without fk? should I insert fk after inserted any listItem on my list?

**Comment by mogaleaf** [comment]
Hoo I think I understand now what you mean by listItem !! It's rows in List table ! so called them List :) In user table we insert user so each row is a user. In List table you insert List so each row is a list.
Now Each list should contains multiple item (that was my misunderstanding with listitem). So you need a Item table and this Item table needs a foreign key on list table. One item is attached to One list.  One list can have multiple items. 
Now each item can have one reminder and one tag to make it simple.

So you need : 
List table primary key id => You have it , foreign key on User table and you have it
Item table primary key id => You need to add it. Item table has a foreign key on List table. you can call it list_id
Reminder table primary key id => You have it. Reminder table has a foreign key on Item Table, you can call it item_id
Tags table primary key id => You have it. Tags table has a foreign key on Task table, you can call it item_id.

Is it better ?

**Comment by Mehmet-Ergorgec** [comment]
I am sorry @mogaleaf now I notice your message because of homework week3. I think there is a conflicts in naming. :) I understand what you mean. Thank you

**Pull request closed without merging**